Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/SendSearchRequestToDrivers/Handle/Internal/DriverPool.hs

```diff
@@ -53,18 +54,18 @@ isBatchNumExceedLimit ::
     CacheFlow m r
   ) =>
   DriverPoolConfig ->
-  Text ->
+  Id DSR.SearchRequest ->
   m Bool
-isBatchNumExceedLimit driverPoolConfig transactionId = do
+isBatchNumExceedLimit driverPoolConfig searchReqId = do
   let maxNumberOfBatches = driverPoolConfig.maxNumberOfBatches
-  currentBatchNum <- getPoolBatchNum transactionId
+  currentBatchNum <- getPoolBatchNum searchReqId
   return $ currentBatchNum >= maxNumberOfBatches
 
-driverPoolKey :: Text -> Text
-driverPoolKey transactionId = "Driver-Offer:DriverPool:transactionId-" <> transactionId
+driverPoolKey :: Id DSR.SearchRequest -> Text
+driverPoolKey searchReqId = "Driver-Offer:DriverPool:SearchReqId-" <> searchReqId.getId
 
-driverPoolBatchKey :: Text -> PoolBatchNum -> Text
-driverPoolBatchKey transactionId batchNum = driverPoolKey transactionId <> ":BatchNum-" <> show batchNum
+driverPoolBatchKey :: Id DSR.SearchRequest -> PoolBatchNum -> Text
+driverPoolBatchKey searchReqId batchNum = driverPoolKey searchReqId <> ":BatchNum-" <> show batchNum
```

Comment:
> Changing keys name can cause cache hit and as there is not fallback for this data it can cause potential. Can you check what could be the impact of this and how much?

Worst thing can happen here - we will get empty batch, which will result in search cancellation. Not a big issue really